enhancements/sig-policy/66-delete-policy-resources/README.md

```diff
@@ -0,0 +1,221 @@
+and that policy contains a `ConfigurationPolicy` with `removeObjectsOnDelete: true`, the
+object will be deleted on the old cluster and created on the new one. This means that
+changing the placement of an `enforce` policy will no longer result in objects
+persisting indefinitely on the old managed cluster.
+
+
+### Risks and Mitigation
+
+The main risk of adding an enhancement like this would be users deleting objects
+that they want to remain on the cluster after they delete a policy. We plan to
+mitigate this risk by making the `spec.removeObjectsOnDelete` field default to
+`false`, so users will not delete anything without having to set that field first.
+
+## Design Details
+
+The `spec.removeObjectsOnDelete` field would look something like this:
+
+```
+spec:
+  namespaceSelector:
+    exclude:
+    - kube-*
+    include:
+    - default
+  object-templates:
+  - complianceType: musthave
+    objectDefinition:
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: proposal-pod
+      spec:
+        containers:
+        - image: nginx:1.18.0
+          name: nginx
+          ports:
+          - containerPort: 80
+  remediationAction: enforce
+  severity: low
+  removeObjectsOnDelete: true //controller will remove proposal-pod
+```
+
+Here is what the `ConfigurationPolicy` status would look like with the new field:
+
+```
+status:
+  compliancyDetails:
+  - Compliant: Compliant
+    Validity: {}
+    conditions:
+    - lastTransitionTime: "2022-06-09T19:27:13Z"
+      message: pods [demo] in namespace default found as specified, therefore this Object template is compliant
+      reason: K8s `must have` object already exists
+      status: "True"
+      type: notification
+  compliant: Compliant
+  lastEvaluated: "2022-06-09T19:27:53Z"
+  lastEvaluatedGeneration: 1
+  relatedObjects:
+  - compliant: Compliant
+    object:
+      apiVersion: v1
+      kind: pods
+      metadata:
+        name: demo
+        namespace: default
+    properties:
+      createdByPolicy: true //delete when parent configPolicy is deleted
+    reason: Resource found as expected
+```
+### Scalability
+The extra fields that will be needed to store data for this change should not impact scalability much, as we
+would only add a few boolean fields to each `ConfigurationPolicy`. Deleting a policy will take a bit longer if
+it is placed on a large amount of managed clusters, since the controller will now have to remove all created
+objects from each managed cluster in addition to the `ConfigurationPolicy`. However, this would still be much
+faster than it would take a user to clean up those resources otherwise - without this change, the easiest
+way for users to clean up resources created by a policy would be to create another policy to delete the
+objects after deleting the original policy, so this could end up helping with scalability because users
+will not have to create as many policies on the hub cluster.
+
+### Open Questions
+
```

Comment:
What behavior do we expect if the user chooses to edit the policy and change `remediationAction: enforce` to `inform`?  If the policy created resources and has `createdByPolicy: true`, it would seem the deletion of an `inform` policy would introduce a change to the cluster which is against the design for how `inform` works.  Should changing to inform disable the cleanup?

Also -- now that we have a cool namespace selector, what if a label on a namespace changes and the policy no longer applies to a particular namespace?  Will we realize our namespace matching is different and the resource needs to be cleaned up -- this is not a policy delete scenario, just a namespace matching change scenario.

Reply:
@gparvin Yeah, for the first question we would only delete the object if the remediationAction is set to enforce. The namespace mismatching is a good point but I don't think it applies to this enhancement since it's not something we need to worry about upon policy deletion. I think that we should open a separate issue to fix resources persisting after a namespace change (we could also include resources that have been renamed persisting with their old name in that second issue, since I think the solution would be similar).